### PR TITLE
Hosted vs. shipped

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: _posts/2018-05-21-shipped-and-hosted-software.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+The Taskcluster team's work on
+[redeployability](http://code.v.igoro.us/posts/2018/01/r14y.html) means
+switching from a *hosted* service to a *shipped* application.
+
+A hosted service is one where the authors of the software are also running the
+main instance of that software. Examples include Github, Facebook, and
+[Mozillians](https://mozillians.org). By contrast, a shipped application is
+deployed multiple times by people unrelated to the software's authors. Examples
+of shipped applications include Gitlab, Joomla, and the Rust toolchain. And, of
```
Comment (imbstack) [inline _posts/2018-05-21-shipped-and-hosted-software.md:16]: You're not going to mention buildbot here? :p

File: _posts/2018-05-21-shipped-and-hosted-software.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+(implemented via [Terraform](http://terraform.io/)) to produce a set of
+[Kubernetes](http://kubernetes.io/) resources for deployment with `kubectl`.
+Kubernetes and Terraform both have limited support for migration from one
+release to another: Terraform will only create or modify changed resources, and
+Kubernetes will perform a phased roll-out of any modified resources.
+
+By the way, all of this build-and-release functionality is implemented in the
+new [taskcluster-installer](https://github.com/taskcluster/taskcluster-installer).
+
+### Service Discovery
```
Comment (imbstack) [inline _posts/2018-05-21-shipped-and-hosted-software.md:107]: Should we talk about inter-service routing without going back out through ingress here or just ignore it for now?

File: _posts/2018-05-21-shipped-and-hosted-software.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+As we transition to using Postgres instead of Azure, we will need to adopt some
+of the common migration tools. Ideally we can support downtime-free upgrades
+like azure-entities does, instead of requiring downtime to run DB migrations
+synchronously. [Bug 1431783](https://bugzilla.mozilla.org/show_bug.cgi?id=1431783)
+tracks this work.
+
+### Customization
+
+As a former maintainer of Buildbot, I've had a lot of experience with CI
```
Comment (imbstack) [inline _posts/2018-05-21-shipped-and-hosted-software.md:160]: Honestly you may want to mention this earlier. It lends a lot of credibility to anything you say in this post!

File: _posts/2018-05-21-shipped-and-hosted-software.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+### Everything is Public!
+
+Mozilla is Open by Design, and so is Taskcluster: with the exception of data
+that must remain private (passwords, encryption keys, and material covered by
+other companies' NDAs), everything is publicly accessible. We take advantage of
+that by reading data without any authentication. For example, the [action
+specification](https://docs.taskcluster.net/manual/using/actions/spec)
+describes downloading a decision task's `public/action.json` artifact. Nowhere
+does it mention providing any credentials to fetch the decision task, nor to
```
Comment (imbstack) [inline _posts/2018-05-21-shipped-and-hosted-software.md]: This kinda makes it feel like _nothing_ is authenticated and that there isn't such a thing as credentials in taskcluster. I like the general point you're making, but maybe call out scopes/creds a bit?

**Review by imbstack — APPROVED** [review]
Awesome.

File: _posts/2018-05-21-shipped-and-hosted-software.md (hunk 3 above)
Comment (djmitche) [reply _posts/2018-05-21-shipped-and-hosted-software.md:160]: I didn't want to overplay my hand :)

File: _posts/2018-05-21-shipped-and-hosted-software.md (hunk 4 above)
Comment (djmitche) [reply _posts/2018-05-21-shipped-and-hosted-software.md]: Sure, good point.

**Pull request merged**